testdoubles/wharfclientapifetcher_mock.go

```diff
@@ -5,44 +5,60 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+// WharfClientAPIFetcherMock is a mock variant of the wharfapi.Client with the
+// helps of github.com/stretchr/testify/mock.
 type WharfClientAPIFetcherMock struct {
 	mock.Mock
 }
 
+// GetTokenById returns a token as identified by its ID.
 func (m *WharfClientAPIFetcherMock) GetTokenById(tokenID uint) (wharfapi.Token, error) {
 	args := m.Called(tokenID)
 	return args.Get(0).(wharfapi.Token), args.Error(1)
 }
 
-func (m *WharfClientAPIFetcherMock) GetToken(token string, userName string) (wharfapi.Token, error) {
+// GetToken returns a token as identified by its token and username strings.
+func (m *WharfClientAPIFetcherMock) GetToken(token, userName string) (wharfapi.Token, error) {
 	args := m.Called(token, userName)
 	return args.Get(0).(wharfapi.Token), args.Error(1)
 }
 
+// PostToken creates a new token and returns the created token,
+// populated with its newly assigned ID.
 func (m *WharfClientAPIFetcherMock) PostToken(token wharfapi.Token) (wharfapi.Token, error) {
 	args := m.Called(token)
 	return args.Get(0).(wharfapi.Token), args.Error(1)
 }
 
+// GetProviderById returns a provider as identified by its ID.
 func (m *WharfClientAPIFetcherMock) GetProviderById(providerID uint) (wharfapi.Provider, error) {
 	args := m.Called(providerID)
 	return args.Get(0).(wharfapi.Provider), args.Error(1)
 }
 
-func (m *WharfClientAPIFetcherMock) GetProvider(providerName string, URLStr string, uploadURLStr string, tokenID uint) (wharfapi.Provider, error) {
-	args := m.Called(providerName, URLStr, uploadURLStr, tokenID)
+// GetProvider returns a provider as identified by its name, URL, and upload
+// URL strings, as well as its token ID reference.
+func (m *WharfClientAPIFetcherMock) GetProvider(providerName, urlStr, uploadURLStr string, tokenID uint) (wharfapi.Provider, error) {
+	args := m.Called(providerName, urlStr, uploadURLStr, tokenID)
 	return args.Get(0).(wharfapi.Provider), args.Error(1)
 }
 
+// PostProvider creates a new provider and returns the created provider,
+// populated with its newly assigned ID.
 func (m *WharfClientAPIFetcherMock) PostProvider(provider wharfapi.Provider) (wharfapi.Provider, error) {
 	args := m.Called(provider)
 	return args.Get(0).(wharfapi.Provider), args.Error(1)
 }
 
+// PutProject creates or updates a project, based on wether the project ID value
+// is non-zero, and returns the created or updated project,
+// populated with its possibly newly assigned ID.
 func (m *WharfClientAPIFetcherMock) PutProject(project wharfapi.Project) (wharfapi.Project, error) {
 	args := m.Called(project)
 	return args.Get(0).(wharfapi.Project), args.Error(1)
 }
+
+// PutBranches resets the list of branches for a project.
```

Comment:
The word `resets` initially made me think that it would clear the list.
Perhaps `sets` would be clearer?

Reply:
It does clear the list. The API endpoint removes all branches from the DB that are not provided in the request, and adds the ones that are missing

Reply:
Resets to me sounds like it doesn't do the second step.
Something like
```suggestion
// PutBranches clears all branches in the database for a project, and then adds the provided branches.
```
maybe?

Reply:
"Replaces"?

Reply:
I think we found our winner 👍